realm/src/androidTest/java/io/realm/RealmAsyncQueryTests.java

```diff
@@ -0,0 +1,1943 @@
+        }
+    }
+
+    // transforming an async query into sync by calling load to force
+    // the blocking behaviour
+    public void testForceLoadAsync() throws Throwable {
+        final CountDownLatch signalCallbackFinished = new CountDownLatch(1);
+        final Throwable[] threadAssertionError = new Throwable[1];
+        final ExecutorService executorService = Executors.newSingleThreadExecutor();
+        executorService.submit(new Runnable() {
+            @Override
+            public void run() {
+                Looper.prepare();
+                Realm realm = null;
+                try {
+                    realm = openRealmInstance("testForceLoadAsync");
+                    Realm.asyncQueryExecutor.pause();
+
+                    populateTestRealm(realm, 10);
+                    final RealmResults<AllTypes> realmResults = realm.where(AllTypes.class)
+                            .between("columnLong", 0, 4)
+                            .findAllAsync();
+
+                    // notification should be called as well
+                    realmResults.addChangeListener(new RealmChangeListener() {
+                        @Override
+                        public void onChange() {
+                            try {
+                                assertTrue(realmResults.isLoaded());
+                                assertEquals(5, realmResults.size());
+                            } catch (AssertionFailedError e) {
+                                threadAssertionError[0] = e;
+                            }
+                            signalCallbackFinished.countDown();
+                        }
+                    });
+
+                    assertFalse(realmResults.isLoaded());
+                    assertEquals(0, realmResults.size());
+
+                    Realm.asyncQueryExecutor.resume();
+                    boolean successful = realmResults.load();
+
+                    assertTrue(successful);
+                    assertTrue(realmResults.isLoaded());
+                    assertEquals(5, realmResults.size());
+
+                    Looper.loop();
+
+                } catch (Exception e) {
+                    e.printStackTrace();
+                    threadAssertionError[0] = e;
+
+                } finally {
+                    if (signalCallbackFinished.getCount() > 0) {
+                        signalCallbackFinished.countDown();
+                    }
+                    if (realm != null) {
+                        realm.close();
+                    }
+                }
+            }
+        });
+
+        // wait until the callback of our async query proceed
+        signalCallbackFinished.await(7, TimeUnit.SECONDS);
+
+        executorService.shutdownNow();
+        if (null != threadAssertionError[0]) {
+            // throw any assertion errors happened in the background thread
+            throw threadAssertionError[0];
+        }
+    }
+
+    // UC:
+    //   1- insert 10 objects
```

Comment:
yes but some steps are events (happening in the Handler or Queue) if we end up annotating some steps & not others, it may create a confusing or a sense of missing steps.